### ISSUE TITLE
libguestfs-tools is broken

### ISSUE THREAD

**Issue by mpilking** [teamdfir/sift#517]
The libguestfs-tools package (virt-filesystems) does not work in the current 20.04 builds of SIFT (e.g. build from 2021-03-31), nor does it work in the publicly available SIFT based on 18.04.  I think we should either fix it before releasing a new Community SIFT, or just remove it.  In general, libguestfs-tools seems too brittle.  Alternatively, qemu-nbd is already in the new SIFT and I confirmed it works for mounting VHDX files. (You can test it using the commands under mount_vhdx.sh in this Gist: https://gist.github.com/allenyllee/0a4c02952bf695470860b27369bbb60d)

So in a nutshell, if libguestfs-tools can be fixed without too much effort, that's great.  If not, I suggest just removing the package so it doesn't ship with a broken install.

**Comment by ekristen** [comment]
What exactly is not working about libguestfs-tools. We are simply installing the pre-build package, so if it's not working, sounds like something is going with the upstream tool which is a bit odd, but not unheard of.

**Comment by mpilking** [comment]
The problem I'm seeing is that when I try to access a .vhdx image with `virt-filesystems`, it fails due to a missing file in the "supermin" mini VM that `virt-filesystems` spins up. It reports an error such as:

`guestfsd: error while loading shared libraries: libtsk.so.19: cannot open shared object file: No such file or directory`

It is not creating the supermin VM properly.  You can first validate that the libguestfs system overall is broken by running `libguestfs-test-tool` and see that it will error out. You should get something like "libguestfs: error: appliance closed the connection unexpectedly, see earlier error messages"

I've read up on the issue and tried several fixes:

- clear out all temp files/supermins via `rm -rf /var/tmp/.guestfs*`
- Copy the missing file into the "supermin" VM. I did that, but then it just fails again on a different file missing. 
- I even tried packaging a supermin VM from the current working 508 SIFT via "libguestfs-make-fixed-appliance --xz" and transferring it to the new SIFT and setting the environment variable "LIBGUESTFS_PATH". The "libguestfs-test-tool" command worked after that, but "virt-filesystems -a" still did not. It failed with a different error.

This issue of missing files in the supermin VM is described in multiple threads here: https://www.mail-archive.com/search?q=guestfsd+no+such+file+error+while+loading+shared&l=libguestfs%40redhat.com. I've gone through these and tried most if not all the fixes--other than rebuild libguestfs from scratch (which is one recommendation from the maintainer Richard Jones in one of the threads).

Overall, I've spent far too much time on it. I don't recommend you spend much time on it. We have a good alternative, which is `qemu-nbd`. I just think we should remove this package rather than keeping it installed when we know it is broken.

**Comment by ekristen** [comment]
Oh, this might be pretty easy to fix, looks like libguestfs-tools is built against a different version of libtsk that is installed. I'll take a look at the source for libguestfs, might be able to solve this in a couple different easy ways.

**Comment by mpilking** [comment]
I don't think it's libtsk that is the problem, even though it initially chokes on that file. I've tried to add that file into the supermin VM and then it fails on a different totally unrelated missing file. I think it more about building the supermin.  Feel free to try a fix, but again, I wouldn't let it take up much of your time. This tool overall seems too brittle.

**Comment by ekristen** [comment]
We've removed libguestfs-tools from the build process for now until we confirm it's fixed and working again.
